packages/api/src/domain/medical/ihe-result.ts

```diff
@@ -14,3 +18,7 @@ export interface PatientDiscoveryResult extends BaseResultDomain {
 export interface DocumentQueryResult extends BaseResultDomain {
   data: DocumentQueryResponse;
 }
+
+export interface DocumentRetrievalResult extends BaseResultDomain {
+  data: DocumentRetrievalResponse;
```

Comment:
I know I approved the other PR that introduced this pattern, but the more I think about it, the more I find this approach to use the SDK's type on the model problematic.

Imagine we need to make a change on the SDK so `DocumentRetrievalResponse` now has a new required field or some other breaking change on the model.

With this approach, we're essentially creating a DB layer that (1) requires a SDK and (2) has it's model defined by an external application.

My suggestion is to define here/API the types used in all `data` fields.

Reply:
I suppose both have their pros and cons because then if we were to define the data types here we can potentially run into issue where we update in one place but then forget to update in another leading to mismatching schemas. Unless you are suggesting to completely move those schemas from the sdk into the api. Which could make sense as i dont image them being used anywhere else. It just made sense to have everything related to the ihe gateway in the sdk.

Reply:
> f we were to define the data types here we can potentially run into issue where we update in one place but then forget to update in another

I thing depending how we build it, that couldn't happen:
- domain/model defined on the API
- SDK has its type/schema
- API's endpoints use the SDK as input
- if types don't match, we get a compilation error on the route when we try to send the shape/type from the SDK into the API/command